### ISSUE TITLE
ElastichSearch cluster with single node failing to start

### ISSUE THREAD

**Issue by salaboy** [elastic/helm-charts#688]
**Chart version:**

7.5.1 + ES version 6.8.5

**Kubernetes version:**

v1.16.8-gke.15

**Kubernetes provider:** E.g. GKE (Google Kubernetes Engine)
GKE

**Helm Version:**
v3.1.2

**`helm get release` output**

```
[elasticsearch-master-0] path: /_cluster/health, params: {wait_for_status=green, timeout=1s}
org.elasticsearch.discovery.MasterNotDiscoveredException: null
	at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction$4.onTimeout(TransportMasterNodeAction.java:262) [elasticsearch-6.8.5.jar:6.8.5]
	at org.elasticsearch.cluster.ClusterStateObserver$ContextPreservingListener.onTimeout(ClusterStateObserver.java:322) [elasticsearch-6.8.5.jar:6.8.5]
	at org.elasticsearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onTimeout(ClusterStateObserver.java:249) [elasticsearch-6.8.5.jar:6.8.5]
	at org.elasticsearch.cluster.service.ClusterApplierService$NotifyTimeout.run(ClusterApplierService.java:564) [elasticsearch-6.8.5.jar:6.8.5]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:681) [elasticsearch-6.8.5.jar:6.8.5]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:830) [?:?]

```

</details>

**Describe the bug:**

Setting one replica to ElasticSearch cluster doesn't work as stated here: https://github.com/elastic/helm-charts/blob/master/elasticsearch/examples/config/values.yaml

**Steps to reproduce:**

1. Use the proposed values in the link above

**Expected behavior:**

 A single replica cluster should be able to be configured.. 

**Any additional context:**

**Comment by salaboy** [comment]
I think this might be related to 6.8.5, 
this can be solved by adding 
```
replicas: 1
minimumMasterNodes: 1
```

**Comment by fatmcgav** [comment]
@salaboy Yeh, so the settings required changed between 6.x and 7.x.

See the `6.8` example here: https://github.com/elastic/helm-charts/blob/6.8/elasticsearch/examples/config/values.yaml#L3-L5

We also recommend matching the major version of the chart to the major version of Elasticsearch you're running. So in this case, install the `6.8.10` chart version.

**Comment by salaboy** [comment]
👍 I wonder then how examples can reflect that? @fatmcgav

**Comment by fatmcgav** [comment]
@salaboy The examples for the `6.8` releases do reflect the correct configuration. However the `master` examples reflect the requirements of the current "default" release version.

So I guess the better question is: "How can we make the `6.8` branch more obvious?"

**Comment by jmlrt** [comment]
Hey @salaboy,

We usually recommand using the [GitHub tag](https://github.com/elastic/helm-charts/tags) of the chart version you are using when checking docs to have documentation and examples accurate.
As the documentation and examples from git branches (especially `master` branch) are evolving in the same time as the chart are evolving.

In your case, as you are using 7.5.1 chart version, that would mean using https://github.com/elastic/helm-charts/tree/7.5.1/elasticsearch/examples

Note that we can't provide examples for every single use case, so charts versions < 7.7.0 were usually providing only one simple example for [6.x version](https://github.com/elastic/helm-charts/tree/7.5.1/elasticsearch/examples/6.x) while all the other examples where using the current released version (so [config](https://github.com/elastic/helm-charts/tree/7.5.1/elasticsearch/examples/config) is now to work with 7.5.1 but not necessarily with 6.x versions).

If your configuration differ from the examples provided you should create your own values based on the existing examples but mostly [chart values description](https://github.com/elastic/helm-charts/tree/7.5.1/elasticsearch#configuration) and [Elasticsearch official doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.5/index.html).

In addition, we changed our [branching model](https://github.com/elastic/helm-charts/blob/master/BREAKING_CHANGES.md#new-branching-model) last months when we released 7.7.0 and 6.8.10. By using releases branches, we are now supporting only deployments where the chart version is equal to Elasticsearch but this also allow us to provide a full example suite for each version instead of the single test for `6.x`versions.

I would recommand you to use [6.8.10](https://github.com/elastic/helm-charts/tree/6.8.10/elasticsearch) for both chart version and Elasticsearch version instead of chart 7.5.1 + Elasticsearch 6.8.5. The 6.8.10 [config example](https://github.com/elastic/helm-charts/tree/6.8.10/elasticsearch/examples/config) is tested and known to work.

In the meantime, we are currently working on improving the doc and process to make the impacts of our new branching model more explicit.
